### PR TITLE
FOUR-13922 Configurable Requests and Tasks columns

### PR DESCRIPTION
**Pull request by caleeli**
## Configurable Requests and Tasks columns
Use `edit-screens` as permission to setup Requests and Tasks permissions.

## Solution
- Use `edit-screens` as permission to setup Requests and Tasks permissions.

## How to Test

1. Create a user with permissions:
- edit-screens
- view-users
- view-groups
2. Login the that User
3. This User will have enabled a gear to configure the Columns of Requests and Tasks List

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-13922

## Code Review Checklist
- [x] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [x] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [x] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [x] This solution fixes the bug reported in the original ticket.
- [x] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [x] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [x] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [x] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [x] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy
ci:next
ci:package-savedsearch:FOUR-13922

.

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-03f8379b37.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-03f8379b37.engk8s.processmaker.net

**Review by boliviacoca — APPROVED** [review]

File: ProcessMaker/Http/Controllers/RequestController.php [hunk 1]
```diff
@@ -64,12 +64,16 @@ public function index($type = null)
...
 
         // Get default Saved search config
-        $defaultSavedSearch = SavedSearch::firstSystemSearchFor(
-            Auth::user(),
-            SavedSearch::KEY_REQUESTS,
-        );
-        if ($defaultSavedSearch) {
-            $defaultColumns = SavedSearchController::adjustColumnsOf($defaultSavedSearch->columns);
+        if (!class_exists(SavedSearch::class)) {
+            $defaultSavedSearch = SavedSearch::firstSystemSearchFor(
```
Comment (agustinbusso) [inline ProcessMaker/Http/Controllers/RequestController.php:68]: When package is not installed is calling to `SavedSearch::firstSystemSearchFor` method, that method is in the package

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-03f8379b37.engk8s.processmaker.net

**Review by devmiguelangel — APPROVED** [review]

**Pull request merged**